contract/pkg/cache/ddc_bucket_contract_cache.go

```diff
@@ -216,28 +220,95 @@ func cacheDurationOrDefault(duration time.Duration, defaultDuration time.Duratio
 	return defaultDuration
 }
 
+func isNodeKeyPresent(nodeKeys []string, nodeKey string) bool {
+	for _, key := range nodeKeys {
+		if key == nodeKey {
+			return true
+		}
+	}
+	return false
+}
+
 func toString(value uint32) string {
 	return strconv.FormatUint(uint64(value), 10)
 }
 
 func (d *ddcBucketContractCached) ClusterCreate(cluster *bucket.NewCluster) (clusterId uint32, err error) {
-	//TODO implement me
-	panic("implement me")
+	clusterId, err = d.ddcBucketContract.ClusterCreate(cluster)
+
+	if err != nil {
+		return 0, err
+	}
+
+	// Clear the corresponding cache since the cluster data has been modified.
+	d.ClearBuckets()
+	d.ClearNodes()
+
+	return clusterId, nil
 }
 
 func (d *ddcBucketContractCached) ClusterAddNode(clusterId uint32, nodeKey string, vNodes [][]bucket.Token) error {
-	//TODO implement me
-	panic("implement me")
+	if clusterId == 0 {
```

Comment:
It is possible to have `0` id for the cluster, currently cluster_id is `u32` type, so the very first cluster will have id that equals to `0`